optimizer_wrapper.rb

```diff
@@ -400,6 +400,21 @@ def self.split_independent_vrp(vrp)
     # - there is multimodal subtours
     return [vrp] if (vrp.vehicles.size <= 1) || vrp.services.empty? || vrp.subtours&.any? # there might be zero services
 
+    # Accumulate the skills of services in a relation of ALL_OR_NONE_RELATIONS type
+    vrp.relations.each{ |relation|
+      next unless Models::Relation::ALL_OR_NONE_RELATIONS.include?(relation.type)
```

Comment:
We did not exactly want to only address "all or none relations" but the relations which force services to be in the same route in way or another.
It includes : `same_route`, `sequence`, `order`, `shipment`

`meetup` expects different vehicles to reach the defined missions at the same time.

Reply:
I am not sure we can do this skill accumulation trick vis-a-vis forcing relations as you suggested because (I remember asking you this a while back that) it is possible to plan only some part of an `order` relation as long as we respect the order but not plan some of the services -- isn't that the case anymore? Also, `same_route` relation applies only to planned services and so it is possible to plan a single one of them. But if we accumulate the skills, we might prevent this possibility.

However, for a `shipment`, we cannot just plan the pickup or just the delivery and they have to be on the same route -- so we don't change the feasible solution space by doing the skill accumulation. I believe it was a joint decision to include `sequence` in the all_or_none list during split_solve refacto -- even though we can skip the tail end of the services -- so I kept the same logic but it is not super appropriate to do so either.

I must have misunderstood `meetup` relation -- effectively if this is the definition, we cannot (and should not) accumulate the skills. It means that we need to find a way to "force" these services and the vehicles that can serve them to stay on the same subproblem (! which is easier said then done !) and keep the skill intact  :( 

In this case, seeing that we immediately need the shipment fix (and the other relations are not in high demand) and that we need a proper over-haul of the split_independent logic -- which is too big to do as a quick hotfix -- I suggest I do the skill accumulation fix only for `shipment` relation and at a "higher", more general location (because it is always correct to do this expansion) and create a ticket for the refacto of split_independent to handle all "forcing" relations in a generic fashion (the way we did for split_solve) in the next sprint. Is that good for you?

Reply:
In fact I just realised, split_independent is as high as it gets but I might do it in vrp.create after shipment expansion unless you have a suggestion for a better location?

Reply:
`same_route`, `order` and `sequence` have the same behavior  
mission at rank `n` should be assigned to allow mission at rank `n+1` to be active in the solution.
So for any mission, if it is assigned all the ancestors/predecessors should be assigned as well.

Do you expect that for a relation with services 1,2 and 3 we only assign services 1 and 2 and let service 3 unassigned eventually because of its skills ? I didn't imagine that at first. But we can transmit the skills from service 1 to service 2 and then from service 2 to service 3.

Have this propagation within vrp.create seems to be the good place

Reply:
I like the idea of accumulating forwards -- that works for `shipment` too (it will not affect the split independent of shipment relations if we accumulate in only one direction)

Reply:
Then instead of doing it for `all_or_none` we can do it for all linking relations.

```ruby    # Relations that link multiple services to be on the same route
    LINKING_RELATIONS = %i[
      order
      same_route
      sequence
      shipment
      same_vehicle
    ].freeze
```

Does this list complete -- at least from the point of view of "forward skill accumulation"? 

I specifically asked from skill accumulation point of view because other relations for which we cannot do skill accumulation (e.g., `meetup`, `min/max_dur/day_lapse`) we still need a refacto